test/integration/test/philips_light_test.js

```diff
@@ -47,48 +49,93 @@ Prepper.makeSuite('Control lights locally',function(){
-        expect(listResponse.body.length).equals(3);
+        lights = Object.keys(listResponse.body);
+        expect(lights.length).equals(3);
         expect(listResponse).to.have.status(200);
       });
     });
 
-    describe('once the lights are known', function(){
+    describe('manipulate lights', function () {
       before(function() {
         // turn off all lights in the simulator
         Prepper.philipshue_server.turnOffLight(1);
         Prepper.philipshue_server.turnOffLight(2);
         Prepper.philipshue_server.turnOffLight(3);
 
-        return chakram.get(serviceListURL)
+        return chakram.put(getURL,getterPayload)
         .then(function(listResponse) {
-          lightinfo = listResponse.body;
+          lights = Object.keys(listResponse.body);
+          expect(lights.length).equals(3);
+          expect(listResponse).to.have.status(200);
         });
       });
 
       // Currently, there is no mapping between the foxbox
-      // issues id and the philips hue id until the tag feature is implemented
-      it('Turn on all lights', function(){
-        return Prepper.turnOnLight(lightinfo, serviceURL, 0)
+      // id and the philips hue id until the tag feature is implemented
+      it('Turn on all lights one by one', function(){
+        var lightID = lights[0].replace('getter','setter');
+        var payload = {'select': {'id': lightID}, 'value': { 'OnOff': 'On' } };
+        
+        return chakram.put(setURL,payload)
         .then(function(cmdResponse) {
-          Prepper.expectLightIsOn(cmdResponse);
-          return Prepper.turnOnLight(lightinfo, serviceURL, 1);
-        })
+         expect(cmdResponse).to.have.status(200);
+         expect(cmdResponse.body[lightID]).equals(null);
+         lightID = lights[1].replace('getter','setter');
+         payload = {'select': {'id': lightID}, 'value': { 'OnOff': 'On' } };
+         return chakram.put(setURL,payload);
+       })
         .then(function(cmdResponse) {
-          Prepper.expectLightIsOn(cmdResponse);
-          return Prepper.turnOnLight(lightinfo, serviceURL, 2);
+          expect(cmdResponse).to.have.status(200);
+          expect(cmdResponse.body[lightID]).equals(null);
+          lightID = lights[2].replace('getter','setter');
+          payload = {'select': {'id': lightID}, 'value': { 'OnOff': 'On' } };
+          return chakram.put(setURL,payload);
         })
         .then(function(cmdResponse) {
-         Prepper.expectLightIsOn(cmdResponse);
+          expect(cmdResponse).to.have.status(200);
+          expect(cmdResponse.body[lightID]).equals(null);
+          expect(Prepper.philipshue_server.areAllLightsOn()).to.be.true;
+          
+          //check all lights are reported to be on
+          return chakram.put(getURL,getterPayload);
+        })
+        .then(function(listResponse) {
+          lights = Object.keys(listResponse.body);
+          expect(lights.length).equals(3);
+          expect(listResponse).to.have.status(200);
```

Comment:
I just realized the 200 checks would be best done before we manipulate the body. For example, the server fails with a 50x: you'll get an error saying that there are no 3 lights, whereas the problem is more serious.